### PR TITLE
dockerfile and startApp.sh files fixed to work on mac and windows

### PR DESCRIPTION
**Pull request by shreeya1510**
changes in both files to work on both architectures - arm64 & amd64

### CODE REVIEW

File: MQTicketService/Dockerfile [hunk 1]
```diff
@@ -10,8 +10,14 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-FROM docker.io/library/maven:3.6-openjdk-11 AS build
+#
```
Comment (chughts) [inline MQTicketService/Dockerfile:13]: copyright date needs to be modified to `# © Copyright 2019, 2024 IBM Corporation`

File: MQTicketService/Dockerfile [hunk 2]
```diff
@@ -10,8 +10,14 @@
...
+# Specify base image and platform for the build
```
Comment (chughts) [inline MQTicketService/Dockerfile]: default build base

File: MQTicketService/Dockerfile [hunk 3]
```diff
@@ -10,8 +10,14 @@
...
+ARG imageName=icr.io/ibm-messaging/mq:latest
```
Comment (chughts) [inline MQTicketService/Dockerfile]: Both are base images, should rename the ARGs so that it is more meaningful. eg. baseImageStep1, and baseImageStep2

Comment (chughts) [reply MQTicketService/Dockerfile]: or baseImageBuildStage, baseImageRunStage

File: MQTicketService/Dockerfile [hunk 4]
```diff
@@ -21,21 +27,46 @@ COPY ./TicketGenerator /data/TicketGenerator
 RUN cd /data/TicketGenerator && \
     mvn -f /data/TicketGenerator/pom.xml clean package;
 #
+# stage 2
+#
 #
 # Switch to the MQ image
-# Tested with
+# Tested with
```
Comment (chughts) [inline MQTicketService/Dockerfile:35]: This isn't true anymore. I think its now 9.4.0.0 only.

File: MQTicketService/Dockerfile [hunk 5]
```diff
@@ -21,21 +27,46 @@ COPY ./TicketGenerator /data/TicketGenerator
...
+# Set platform architecture as an environment variable
+ENV platformArch=${platformArch}
+#
 USER 0
-
-RUN curl -Lo OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz && \
+#
+#
+RUN if [ "$platformArch" == "amd64" ]; then \
+    curl -Lo OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz && \
```
Comment (chughts) [inline MQTicketService/Dockerfile]: paramatize `jre_x64_linux_hotspot_11.0.13_8` and `11.0.13_8` so they aren't repeated, and only set once.

Comment (chughts) [reply MQTicketService/Dockerfile]: rather like the arm64 version

File: MQTicketService/Dockerfile [hunk 6]
```diff
@@ -48,3 +79,22 @@ RUN chmod a+x /data/startApp.sh
 # Run MQ and execute TicketGenerator
 USER 1001
 ENTRYPOINT [ "/data/startApp.sh" ]
+
```
Comment (chughts) [inline MQTicketService/Dockerfile]: Remove extra lines at the end of the file

File: MQTicketService/startApp.sh [hunk 7]
```diff
@@ -20,4 +20,14 @@ sleep 30
 # Set environment
 . /opt/mqm/bin/setmqenv -s
 # Execute TicketGenerator
-/jdk-11.0.13+8-jre/bin/java -cp /data/TicketGenerator/target/TicketGenerator-1.4.jar com.ibm.mq.badge.Manager
\ No newline at end of file
+
```
Comment (chughts) [inline MQTicketService/startApp.sh:23]: Update copyright to `Copyright 2019, 2024 IBM Corporation`

**Review by chughts — CHANGES_REQUESTED** [review]

**Review by chughts — APPROVED** [review]

**Review by chughts — CHANGES_REQUESTED** [review]
The readme needs to be updated to show how to pass in the images (and in the case of ARM64, how to create an image), used by the docker file.

File: MQTicketService/Dockerfile [hunk 8]
```diff
@@ -11,13 +11,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# Specify base image and platform for the build
-ARG imageName=icr.io/ibm-messaging/mq:latest
-ARG platName=docker.io/library/maven:3.6-openjdk-11
+# Specify default base images 
+ARG baseImageRunStage=icr.io/ibm-messaging/mq:latest
+ARG baseImageBuildStage=docker.io/library/maven:3.6-openjdk-11
```
Comment (rcoppen) [inline MQTicketService/Dockerfile:16]: Can this be pre-canned for arm64 builds to save the user the effort of going and working out what is needed. We do this for amd64 already.

File: MQTicketService/Dockerfile [hunk 9]
```diff
@@ -47,10 +46,11 @@ USER 0
 #
 #
 RUN if [ "$platformArch" == "amd64" ]; then \
```
Comment (rcoppen) [inline MQTicketService/Dockerfile:51]: `platformArch` is overloaded as both an ARG and an ENV, can they be individually named? e.g., `argPlatformArch` and `envPlatformArch`

File: MQTicketService/Dockerfile [hunk 10]
```diff
@@ -11,13 +11,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# Specify base image and platform for the build
-ARG imageName=icr.io/ibm-messaging/mq:latest
-ARG platName=docker.io/library/maven:3.6-openjdk-11
+# Specify default base images 
+ARG baseImageRunStage=icr.io/ibm-messaging/mq:latest
```
Comment (rcoppen) [inline MQTicketService/Dockerfile]: User needs to know to specify this if they want to build on arm64 based on the MQ image they built. Is this included in a README?

File: MQTicketService/Dockerfile [hunk 11]
```diff
@@ -47,10 +46,11 @@ USER 0
...
-    curl -Lo OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz && \
-    ls -l OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz && file OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz && \
-    tar -xvzf OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz; \
+    curl -Lo OpenJDK11Ux64.tar.gz https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jre_x64_linux_hotspot_11.0.13_8.tar.gz && \
+    ls -l OpenJDK11Ux64.tar.gz && file OpenJDK11Ux64.tar.gz && \
+    tar -xvzf OpenJDK11Ux64.tar.gz; \
 	fi
+# 
 #
 RUN if [ "$platformArch" == "arm64" ]; then \
```
Comment (rcoppen) [inline MQTicketService/Dockerfile]: The two if statements deal with amd64 and arm64, but it's not clear what should happen if the value is set to something else e.g., 'typo64'. Ideally we should check for that and fail. One option would be to pre-check and make sure the two `if` branches are safe or add an if, elif and else.

**Review by rcoppen — CHANGES_REQUESTED** [review]
@shreeya1510 looks good, some minor changes in review comments.

**Review by rcoppen — APPROVED** [review]

File: MQTicketService/startApp.sh [hunk 12]
```diff
@@ -20,4 +20,14 @@ sleep 30
...
+if [ "$envPlatformArch" == "amd64" ]
+then
+    /jdk-11.0.13+8-jre/bin/java -cp /data/TicketGenerator/target/TicketGenerator-1.4.jar com.ibm.mq.badge.Manager 
+elif [ "$envPlatformArch" == "arm64" ] 
+then
+    /jdk-11.0.22+7-jre/bin/java -cp /data/TicketGenerator/target/TicketGenerator-1.4.jar com.ibm.mq.badge.Manager 
+ else 
+    echo "Error: Valid platform architectures are amd64 or arm64" 
+    exit 1
```
Comment (chughts) [inline MQTicketService/startApp.sh:32]: Was this tested to see what happens if neither `amd64` or `arm64` have been set.

We are running a command inside a docker build. The run will exit, but not the docker build

**Review by chughts — CHANGES_REQUESTED** [review]
Not sure about the exit in the platform else clause, as it won't abort the build. A good series of visible error messages may be all that is needed.

**Review by chughts — APPROVED** [review]

**Pull request merged**